### DISCUSSION TITLE
Question regarding wLAN inside car

### DISCUSSION THREAD

**Discussion by ks0002544** [tesla-android/issue-tracker#222] [General]
So, I have the following question. Pi4 with Android installed has possibility only WiFi hotspot. And it is written by the developer. So I can not switch off wifi hotspot and swith on to ordinary wifi. It looks as Pi4 is server (or router) which sets ip adrr to all units (tesla, my phone etc.). Imagine if i have mobile router (with akkum for 6 hours - for example tplink) in tesla and this router sets ip adrr to tesla and Pi4. Only ip adrrs must not be in range 192.168.x. It will be smal wLAN inside car. It will work? I can do it myself but I need ordinary wifi on my Pi4

**Comment by mikegapinski** [comment]
It is supported already. If you plug in your router to the pi via Ethernet just make sure to change the dhcp range to something in the public space.

**Reply by ks0002544** [reply]
but I need to do it not in wired mode but in wireless. I can not switch off hotspot - it switch on all time. Is it possible to make choice between hotspot or ordinary wifi. I found switcher of wifi and I can see my home wifi net but no chance to connect.

**Reply by mikegapinski** [reply]
No, the hotspot is designed to stay on all the time.

There is no other way to interact with this device

**Reply by ks0002544** [reply]
the main question for me is - sim card of mobile operator. At the beggining it was sim card for tesla modem - works bad, then you proposed to use LTE  modem - once again need sim card. Also sim card in my phone. And I need to check payment status of my all sim cards. Why I can not use my smartphone mobile for internet surf?

**Reply by mikegapinski** [reply]
You can. Just connect it with a usb cable to the pi. It will use usb tethering just fine

**Reply by ks0002544** [reply]
yes I do it as you told but for this connection i need more power supply from 12V socket

**Reply by ks0002544** [reply]
I have old Pi3 with android installed (10 probably). I can use it or wifi hotspot or ordinary wifi.

**Reply by ks0002544** [reply]
> No, the hotspot is designed to stay on all the time.
> 
> There is no other way to interact with this device

What is the reason for this limitation?

**Reply by mikegapinski** [reply]
How are you supposed to use Tesla Android if it wouldn't broadcast its Wi-Fi network for the car to connect to. There is no other way
